tests/functional/object/mcg/test_noobaa_use_new_internal_cert_after_rotation.py

```diff
@@ -0,0 +1,116 @@
+from datetime import datetime
+import logging
+
+from ocs_ci.ocs.constants import SECRET, NOOBAA_S3_SERVING_CERT
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.framework.pytest_customization.marks import (
+    tier1,
+    polarion_id,
+    bugzilla,
+    skipif_ocs_version,
+    red_squad,
+    mcg,
+    config,
+)
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.pod import get_noobaa_endpoint_pods
+from ocs_ci.utility.retry import retry
+
+
+logger = logging.getLogger(__name__)
+
+
+@retry(UnexpectedBehaviour, tries=10, delay=3, backoff=1)
+def get_validity_time_of_certificate(noobaa_endpoint_pods, cmd, old_validity=None):
```

Comment:
You can split this function into two make it more neater. Also the doc string for this function needs correction.

```suggestion
def get_validity_time_of_certificate(noobaa_endpoint_pods):
    cmd = (
            "openssl s_client -connect localhost:6443 -showcerts 2>/dev/null </dev/null | sed -ne "
            "'/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' | openssl x509 -text -noout"
        ) 
    new_cmd_output = noobaa_endpoint_pods[0].exec_sh_cmd_on_pod(command=cmd)
    logger.info("Certificate validity info: {new_cmd_output}")
    # Get the validity time of certificate
    cmd_output = new_cmd_output.split(",")
    cmd_output = cmd_output[0].split("\n")
    new_validity = cmd_output[7].split(",")[0] + cmd_output[8].split(",")[0]
    return new_validity

@retry(UnexpectedBehaviour, tries=10, delay=3, backoff=1)
def verify_cert_validity(noobaa_endpoint_pods, old_validity):
     new_validity= get_validity_time_of_certificate(noobaa_endpoint_pods)
     if new_validity == old_validity:
        logger.warn("New certificate is not created \n New validity: {new_validity}\n Old validity: {old_validity}")
        raise UnexpectedBehaviour
     return True
```

Something like above.

Reply:
done